test/ir-to-ir/Tests/LiftProgramLambdasSpec.hs

```diff
@@ -0,0 +1,84 @@
+module Tests.LiftProgramLambdasSpec where
+
+import qualified IR.IR                         as I
+import qualified IR.Types.Classes              as C
+
+import IR.LambdaLift
+import           Common.Pretty                  ( spaghetti )
+import           Common.Compiler                ( runPass )
+import           Front.Ast
+import           Front.Parser                   ( parseProgram )
+import           Front.ParseOperators           ( parseOperators )
+import           IR.LowerAst                    ( lowerProgram )
+import IR.ClassInstantiation (instProgram)
+import IR.Types.Poly as Poly
+import           IR.TypeInference               ( inferExpr
+                                                , inferProgram
+                                                )
+
+import           Control.Comonad                ( Comonad(..) )
+import           Data.Bifunctor                 ( Bifunctor(second) )
+import           Test.Hspec                     ( Spec(..)
+                                                , it
+                                                , pending
+                                                , shouldBe
+                                                , describe
+                                                )
+
+lowerAndLift :: Either String Program -> Either String (I.Program Poly.Type)
+lowerAndLift (Left e) = Left "Failed to parse program"
+lowerAndLift (Right p) =
+  case runPass $ lowerProgram (parseOperators p) >>= inferProgram >>= instProgram >>= liftProgramLambdas of
+    Left e' -> Left $ show e'
+    Right p' -> Right p'
+
+spec :: Spec
+spec = do
+
+  it "lifts a lambda without free variables" $ do
+    let nonLiftedProgram = parseProgram $ unlines
+          [ "bar: Int = 5"
+          , "baz x: Int -> Int ="
+          , "  x + 1"
+          , "foo y: Int -> Int ="
+          , "  let adder (z: Int) -> Int = z + 1"
+          , "  adder y"
+          ]
+        liftedProgram = parseProgram $ unlines
+          [ "bar: Int = 5"
+          , "baz x: Int -> Int ="
+          , "  x + 1"
+          , "anon0 z: Int -> Int ="
+          , "  z + 1"
+          , "foo y: Int -> Int ="
+          , "  let adder = anon0"
+          , "  adder y"
+          ]
+        nestedToLifted = lowerAndLift nonLiftedProgram
+        liftedToLifted = lowerAndLift liftedProgram
+    nestedToLifted `shouldBe` liftedToLifted
```

Comment:
Suggestion, though this might be more effort than it's worth: when we compare the two programs, we don't really want to compare the mangled names. Not a blocker for this PR, but it suggests that we should have some kind of more sophisticated IR comparison mechanisms modulo alpha renaming.

Reply:
Opened issue #61